src/view/context/view.cpp

```diff
@@ -1,19 +1,60 @@
 #include "view/context/view.hpp"
 #include "mainwindow.hpp"
 
-Context::View::View(lib::spt::api &spotify, spt::Current &current,
+Context::View::View(lib::spt::api &spotify, lib::settings &settings, spt::Current &current,
 	const lib::cache &cache, QWidget *parent)
-	: QDockWidget(parent)
+	: QDockWidget(parent),
+	spotify(spotify),
+	current(current),
+	cache(cache)
 {
-	content = new Context::Content(spotify, current, cache, this);
-	setWidget(content);
+	// check if user wants a scalable album image or not
+	albumShouldBeExpandable = settings.general.expand_album_cover;
+	if (albumShouldBeExpandable)
+	{
+		horizContent = new Context::HorizContent(spotify, current, cache, this);
+		setWidget(horizContent);
+	}
+	else 
+	{
+		content = new Context::Content(spotify, current, cache, this);
+		setWidget(content);
+	}
 
 	title = new Context::Title(spotify, current, cache, this);
 	setTitleBarWidget(title);
 
 	setFeatures(QDockWidget::DockWidgetMovable | DockWidgetFloatable);
 }
 
+void Context::View::reloadAlbumContent(bool shouldBeExpandable)
+{
+	albumShouldBeExpandable = shouldBeExpandable;
+	if (content != nullptr)
+	{
+		content->remove();
```

Comment:
You don't have to call `QObject::deleteLater()` from within the instance itself. It's like you're overriding the method, but just call the base method anyway, not really doing anything. `content->deleteLater()` does the same thing.

Reply:
Oh DUH! Yea that's a stoopid miss on my end :)